feat(datasource): add Deno datasource (#18878)
renovatebot/renovate: lib/modules/datasource/deno/readme.md

> This datasource extracts dependencies from a [denoland/apiland](https://github.com/denoland/apiland) compatible registry server.

Reviewer: ```suggestion
This datasource extracts dependencies from a [`denoland/apiland`](https://github.com/denoland/apiland) compatible registry server.
```

> By default, the official Deno registry is used, but this can be overwritten using `registryUrl`.

Reviewer: ```suggestion

By default, Renovate uses the official Deno registry.
You can override this by setting your own `registryUrl`.
```
